File: windows/src/GoalMaker.Core/About/AppInfo.cs
using GoalMaker.Core.Backend;

namespace GoalMaker.Core.About;

/// <summary>
/// Facts about this build, shown in Settings → About. <see cref="LocalOnly"/> is a dev build that neither
/// signs in nor syncs and keeps everything on this PC (docs/sign-in.md).
/// </summary>
public sealed record AppInfo(
    string Version,
    bool IsDevBuild,
    BackendEnvironment Backend,
    BackendEnvironment DefaultBackend,
    bool LocalOnly = false);

File: android/app/src/main/kotlin/com/goalmaker/app/application/about/AppInfo.kt
package com.goalmaker.app.application.about

import com.goalmaker.app.application.environment.BackendEnvironment

/**
 * Facts about this build, shown in Settings → About. [localOnly] is a dev build that neither signs in
 * nor syncs and keeps everything on the device (docs/sign-in.md).
 */
data class AppInfo(
    val versionName: String,
    val isDevBuild: Boolean,
    val backend: BackendEnvironment,
    val defaultBackend: BackendEnvironment,
    val localOnly: Boolean = false,
)
